public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/DimensionFields.tsx

```diff
@@ -33,12 +89,40 @@ const DimensionFields: React.FC<DimensionFieldsProps> = ({ query, dimensionOptio
     onQueryChange(setDimensionFilterValue(query, filterIndex, fieldName, value));
   };
 
-  const onFilterInputChange = (index: number, ev: React.FormEvent) => {
-    if (ev.target instanceof HTMLInputElement) {
-      onFieldChange(index, 'filter', ev.target.value);
+  const onFilterInputChange = (index: number, v: SelectableValue<string>) => {
+    if (v) {
+      onFieldChange(index, 'filter', v.value);
+    } else {
+      onFieldChange(index, 'filter', '');
```

Comment:
Is this necessary? I think the if else condition can be replaced with `v && onFieldChange(index, 'filter', v.value);`

Reply:
Have replaced this with ```  const onFilterInputChange = (index: number, v: SelectableValue<string> | null) => {
    onFieldChange(index, 'filter', v?.value ?? '');
  };```

It's necessary as a null value is passed if the filters are cleared so we handle that accordingly.